controllers/validation/account_validation_controller.go

```diff
@@ -439,10 +452,91 @@ func (r *AccountValidationReconciler) Reconcile(ctx context.Context, request ctr
 			}
 		}
 	}
+	// Updates Below:
+	awsRegion := config.GetDefaultRegion()
+	awsSetupClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, awsclient.NewAwsClientInput{
+		SecretName: utils.AwsSecretName,
+		NameSpace:  awsv1alpha1.AccountCrNamespace,
+		AwsRegion:  awsRegion,
+	})
+
+	// Need to assume role into the cluster account
+	roleToAssume := acct.GetAssumeRole(&account)
+	awsAssumedRoleClient, _, err := acct.AssumeRole(&acct.AccountReconciler{}, reqLogger, &account, awsSetupClient, "", roleToAssume, "")
+	if err != nil {
+		reqLogger.Error(err, "Could not impersonate AWS account", "aws-account", account.Spec.AwsAccountID)
+		return utils.RequeueWithError(err)
+	}
+	// Get a list of regions enabled in the current account
+	regionsEnabledInAccount, err := awsAssumedRoleClient.DescribeRegions(&ec2.DescribeRegionsInput{
+		AllRegions: aws.Bool(false),
+	})
+	if err != nil {
+		// Retry on failures related to the slow AWS API
+		if aerr, ok := err.(awserr.Error); ok {
+			if aerr.Code() == "OptInRequired" {
+				return utils.RequeueWithError(nil)
+			}
+		}
+		reqLogger.Error(err, "Failed to retrieve list of regions enabled in this account.")
+		return utils.RequeueWithError(err)
+	}
+
+	// check if account belongs to accountpool
+	if !account.IsBYOC() {
+		if account.Spec.RegionalServiceQuotas != nil {
+			if account.Status.RegionalServiceQuotas == nil {
+				acct.SetCurrentAccountServiceQuotas(reqLogger, &account, regionsEnabledInAccount)
+			} else {
+				_, err := r.getServiceQuotaRequest(reqLogger, &account, awsSetupClient)
```

Comment:
You're not doing anything with the return values in this function, even though they're reconcile.Result - you can probably just return directly